packages/app/src/app/hooks/useDismissible.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffects } from 'app/overmind';
+
+/**
+ * Localstorage keys used for dismissible modals and banners.
+ */
+type DismissibleKeys =
+  // TODO - Any key in this type should refer to a dismissible item. Right
+  // now there are none yet.
+  'TODO';
+
+type Dismissibles = Partial<Record<DismissibleKeys, true>>;
+
+export const useDismissible = () => {
+  const { browser } = useEffects();
+  const dismissibles = browser.storage.get<Dismissibles>('DISMISSIBLES');
```

Comment:
wouldn't it be safer for each instance of `useDismissible` to keep track of a single key that it could get as a param? so that two instances of the hook will not have different values inside this `dismissibles` reference?

Reply:
then there's no need for a `isDismissed` function, as the hook returns the boolean if the key passed as a param is dismissed or not

Reply:
You mean use it like this?
```tsx
const { isDismissed: isSomethingDismissed, dismiss: dismissSomething } = useDismissible('SOMETHING');
```

Reply:
> so that two instances of the hook will not have different values inside this dismissibles reference?

I think with `localStorage` this will always be a potential problem, right? When two of the same hooks are referencing the same key but for one hook it's dismissed, it doesn't re-render the other hooks and those values will still be the same. Not sure if this will ever be the case with dismissibles though?

Reply:
yes like this. not bulletproof indeed, but most likely useDismissible("key") will only be used once, whereas useDismissible() without any argument will have the reference to all the keys in parallel

Reply:
@olarclara what do you think?

Reply:
@tristandubbeld I like @alexnm's suggestion.